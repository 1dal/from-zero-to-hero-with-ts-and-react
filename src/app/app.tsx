import React, { Component } from 'react'

export type AppProps = {
  hide?: boolean
}
export type AppState = typeof initialState
const initialState = {
  on: false,
  input: '',
  mainColor: 'blue' as 'blue' | 'red',
  lifeCycle: ''
}
export class App extends Component<AppProps, AppState> {
  state = initialState

  private readonly toggleOnState = () =>
    this.setState((state) => ({ on: !state.on }))

  handleStrings(value: string) {
    return value === 'Hello World' ? true : false
  }
  componentWillReceiveProps() {
    this.setState({ lifeCycle: 'componentWillReceiveProps' })
  }
  render() {
    return (
      <div className="App">
        <header className="App-header">
          <h1 className="App-title">Welcome to React</h1>
          <h3 className={this.state.mainColor}>Everyone is Welcome!</h3>
        </header>
        <p className="App-intro">Hello</p>
        <p className="button-state">{this.state.on ? 'Yes!' : 'No!'}</p>
        <button onClick={this.toggleOnState}>Click</button>
        <h2>{this.state.input}</h2>
        <input
          onChange={(e) => this.setState({ input: e.currentTarget.value })}
          type="text"
        />
        <p className="lifeCycle">{this.state.lifeCycle}</p>
      </div>
    )
  }

  componentDidMount() {
    // console.log('CDM!')
    this.setState({ lifeCycle: 'componentDidMount' })
  }
}

export type LinkProps = {
  address: string
  hide?: boolean
}
export class Link extends Component<LinkProps> {
  render() {
    return this.props.hide ? null : <a href={this.props.address}>Click</a>
  }
}
