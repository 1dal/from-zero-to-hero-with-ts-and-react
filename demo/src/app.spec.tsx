import React from 'react';
import { mount } from 'enzyme';
import { App } from './app';

describe(`<App/>`, () => {
  it(`should render`, () => {
    const wrapper = mount(<App/>)

    expect(wrapper).toMatchSnapshot()
  })

  it(`should work as a whole`, () => {
    const wrapper = mount(<App/>)
    const incBtn = wrapper.find('button').at(0)
    const decBtn = wrapper.find('button').at(1)

    expect(wrapper.find('h1').text()).toBe('Hello World')
    expect(wrapper.find('p').text()).toBe('0')

    incBtn.simulate('click')
    incBtn.simulate('click')
    incBtn.simulate('click')
    decBtn.simulate('click')

    expect(wrapper.find('p').text()).toBe('2')

    decBtn.simulate('click')

    expect(wrapper.find('p').text()).toBe('1')
  })
})
