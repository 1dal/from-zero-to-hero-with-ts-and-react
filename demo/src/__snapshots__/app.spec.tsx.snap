// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<App/> should render 1`] = `
<App>
  <div>
    <Component
      message="Hello World"
    >
      <h1>
        Hello World
      </h1>
    </Component>
    <Counter>
      <p>
        0
      </p>
      <button
        onClick={[Function]}
      >
        Increment
      </button>
      <button
        onClick={[Function]}
      >
        Decrement
      </button>
    </Counter>
  </div>
</App>
`;
